File: AmirImamTask.Entities/Entities/TransactionDetail.cs
namespace AmirImamTask.Entities;

[Table(nameof(TransactionDetail))]
public class TransactionDetail : EntityBase
{
    public Guid? TransactionId { get; set; }
    public Transaction? Transaction { get; set; }

    public Guid? ItemId { get; set; }
    public Item? Item { get; set; }
    public int TransactionFactor { get; set; }
    public double Quantity { get; set; }
    public double Price { get; set; }
    public double Value { get; set; }
}

File: AmirImamTask.AngularUI/src/app/models/TransactionDetail.ts
import { Guid } from "guid-typescript";

export class TransactionDetail {
    public Id: string = Guid.EMPTY;
    public TransactionId?: string;
    public ItemId?: string;
    public TransactionFactor: number = 0;
    public Quantity: number = 0;
    public Price: number = 0;
    public Value: number = 0;

    public ItemName?: string;
    public ItemCode?: string;
    public ItemDescription?: string;
}